src/test/compile-fail/dst-bad-coercions.rs

```diff
@@ -23,11 +23,13 @@ pub fn main() {
     let x: *const S = &S;
     let y: &S = x; //~ ERROR mismatched types
     let y: &T = x; //~ ERROR mismatched types
+                   //~^ ERROR the trait bound `*const S: T` is not satisfied
 
     // Test that we cannot convert from *-ptr to &S and &T (mut version)
     let x: *mut S = &mut S;
     let y: &S = x; //~ ERROR mismatched types
     let y: &T = x; //~ ERROR mismatched types
+                   //~^ ERROR the trait bound `*mut S: T` is not satisfied
```

Comment:
These test cases also don't seem to be reflected by the content of the patch, which doesn't add these errors.